initial docs for sim overhaul [WIP] (#283)
PhotonVision/photonvision-docs: source/docs/programming/photonlib/simulation-deprecated.rst

> Simulation Support in PhotonLib Deprecated

Reviewer: Depracated in parenthesis

==========================================

What Is Supported?
------------------

> PhotonLib supports simulation of a Photon Vision camera and processor moving about a field on a robot.

Reviewer: ```suggestion
PhotonLib supports simulation of a camera and coprocessor running PhotonVision moving about a field on a robot.
```

> You can use this to help validate your robot code's behavior in simulation without special wrappers or additional hardware.

Reviewer: ```suggestion
You can use this to help validate your robot code's behavior in simulation without needing a physical robot.
```

Simulation Vision World Model
-----------------------------

Sim-specific classes are provided to model sending one frame of a camera image through PhotonVision. Based on what targets are visible, results are published to NetworkTables.

While processing, the given robot ``Pose2d`` is used to analyze which targets should be in view, and determine where they would have shown up in the camera image.

Targets are considered in view if:

1) Their centroid is within the field of view of the camera.
2) The camera is not in driver mode.
3) The target's in-image pixel size is greater than ``minTargetArea``
4) The distance from the camera to the target is less than ``maxLEDRange``

.. warning:: Not all network tables objects are updated in simulation. The interaction through PhotonLib remains the same.  Actual camera images are also not simulated.

Latency of processing is not yet modeled.

.. image:: images/SimArchitecture-deprecated.svg

Simulated Vision System
-----------------------

> A ``SimVisionSystem`` represents the camera, coprocessor, and PhotonVision software moving around on the field.

Reviewer: ```suggestion
A ``SimVisionSystem`` represents the camera and coprocessor running PhotonVision moving around on the field.
```

> It requires a number of pieces of configuration to accurately simulate your physical setup. Match them to your configuration in PhotonVision, and to your robot's physical dimensions.

Reviewer: Be more specific

.. tab-set-code::

     .. rli:: https://raw.githubusercontent.com/PhotonVision/photonvision/80e16ece87c735e30755dea271a56a2ce217b588/photonlib-java-examples/simaimandrange/src/main/java/frc/robot/sim/DrivetrainSim.java
        :language: java
        :lines: 73-93

> After declaring the system, you should create and add one ``SimVisionTarget`` per target on the field you are attempting to detect.

Reviewer: " per target on the field you are attempting to detect." is ambiguous. is it the field you are trying to detect or the target?

.. tab-set-code::

      .. rli:: https://raw.githubusercontent.com/PhotonVision/photonvision/80e16ece87c735e30755dea271a56a2ce217b588/photonlib-java-examples/simaimandrange/src/main/java/frc/robot/sim/DrivetrainSim.java
        :language: java
        :lines: 95-111

Finally, while running the simulation, process simulated camera frames by providing the robot's pose to the system.

.. tab-set-code::

      .. rli:: https://raw.githubusercontent.com/PhotonVision/photonvision/80e16ece87c735e30755dea271a56a2ce217b588/photonlib-java-examples/simaimandrange/src/main/java/frc/robot/sim/DrivetrainSim.java
         :language: java
         :lines: 138-139

This will cause most NetworkTables fields to update properly, representing any targets that are in view of the robot.

> Robot software which uses PhotonLib to interact with a camera running PhotonVision should work the same as though a real camera was hooked up and active.

Reviewer: Sentence is confusing, and should amke more clear how simulated behavior should match real life camera behavior (you never mention simulatin in this sentence

Raw-Data Approach
-----------------

> Advanced users may wish to directly provide target information based on an existing detailed simulation.

Reviewer: Don't specify advanced users

Simulation Support in PhotonLib (Deprecated)
============================================

> .. attention:: This page details the legacy simulation support. For current Java simulation support, see :doc:`/docs/programming/photonlib/simulation`. 

Reviewer 2: Proposal: replace "legacy" with "pre-2024"

> .. attention:: This page details the legacy simulation support. For current Java simulation support, see :doc:`/docs/programming/photonlib/simulation`. 

Reviewer 2: Would a call to action make sense here? IE "You will need to migrate to the new simulation support"?
